### PR TITLE
Extract common JDBC configurations into their own fragment

### PR DESCRIPTION
**Pull request by hashhar**
Split jdbc-type-mapping fragment into two. One for common JDBC connector configs and other for type mapping.

This will give a place to document common JDBC connector configurations as a follow-up.

### CODE REVIEW

**Comment by hashhar** [comment]
There are two versions of wording for the non-transactional inserts - https://github.com/trinodb/trino/pull/8643/commits/2af398193e0c84c985a4a42d342633ecc48bfd64 and https://github.com/trinodb/trino/pull/8643/commits/2b6c52dcc5545a08b56073f812c0436dd4c5f994. Please advise which one should be kept.

**Comment by findepi** [comment]
> we shouldn't document things which have sensible defaults and generally don't need to be changed and exist only as a escape hatch.

i agree

**Comment by mosabua** [comment]
Fyi @jhlodin and @Jessie212

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 1]
```diff
@@ -46,6 +46,8 @@ catalog for each server. To add another catalog:
 For example, if you name the property file ``sales.properties``, Trino uses the
 configured connector to create a catalog named ``sales``.
 
+.. include:: jdbc-common-configurations.fragment
+
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/clickhouse.rst:50]: Would suggest moving this section above "Multiple ClickHouse servers", or around line 36. Configuration properties are more generally applicable than the above section.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 2]
```diff
@@ -0,0 +1,47 @@
+General configuration properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:3]: Needs an introductory sentence, something like "The following table describes general configuration properties for the connector:"

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+.. list-table::
+  :widths: 30, 40, 30
+  :header-rows: 1
+
+  * - Property name
+    - Description
+    - Default value
+  * - ``case-insensitive-name-matching``
+    - Support case insensitive database and collection names
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:12]: "Enable support for..."

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  * - ``case-insensitive-name-matching.cache-ttl``
+    -
+    - 1 minute
+  * - ``metadata.cache-ttl``
+    - Duration for which metadata, including table and column statistics, is
+      cached
+    - 0 (disabled caching)
+  * - ``metadata.cache-missing``
+    - Cache the fact that metadata, including table and column statistics, is
+      not available
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:23]: This sentence doesn't make clear what this does or when it's needed/beneficial. Does it enable/disable metadata altogether? If not, when would you need to simply cache that it's available?

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - False
+  * - ``allow-drop-table``
+    - Allow connectors to drop tables
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:26]: "Connectors" makes this sound like this one setting will apply to every connector. Would suggest changing to "the connector" which implies it's only applicable for this connector, without naming it.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    - False
+  * - ``join-pushdown.enabled``
+    - Enable join pushdown
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:29]: Prefer "pushdown of join operations", but up to you

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    - False
+  * - ``aggregation-pushdown.enabled``
+    - Enable aggregation pushdown
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:32]: Prefer "pushdown of aggregation operations", but up to you

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    - True
+  * - ``topn-pushdown.enabled``
+    - Enable TopN pushdown
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:35]: Prefer "pushdown of TopN operations", but up to you

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    - Default value
+  * - ``case-insensitive-name-matching``
+    - Support case insensitive database and collection names
+    - False
+  * - ``case-insensitive-name-matching.cache-ttl``
+    -
+    - 1 minute
+  * - ``metadata.cache-ttl``
+    - Duration for which metadata, including table and column statistics, is
+      cached
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:19]: In what increment, minutes? Seconds?

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    - Enable aggregation pushdown
+    - True
+  * - ``topn-pushdown.enabled``
+    - Enable TopN pushdown
+    - True
+  * - ``write.batch-size``
+    - Maximum number of rows to insert in a single batch
+    - 1000
+  * - ``insert.non-transactional-insert.enabled``
+    - Controls how inserts are performed. By default, data insertion is
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:41]: Should be "Control"

**Review by jhlodin — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 3 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:12]: case-insensitive

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 4 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:23]: Where do you find the recorded fact that it went missing?

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+      performed by writing data to a temporary table. Enabling this property
+      causes writes to happen directly to the target table.
+      In rare cases when exceptions occur during the insert operation, data in
+      the target table can be corrupted. Since transactions have been disabled
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:45]: disabled, no

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment [hunk 12]
```diff
@@ -22,3 +22,6 @@ General configuration properties
     - Cache the fact that metadata, including table and column statistics, is
       not available
     - False
+  * - ``allow-drop-table``
```
Comment (electrum) [inline docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:25]: We are going to remove this: https://github.com/trinodb/trino/pull/588

**Comment by jhlodin** [comment]
FYI: At @electrum 's request, Cherry-picked the JDBC type mapping fragments commit into its own PR here - https://github.com/trinodb/trino/pull/8727

**Review by rosewms — COMMENTED** [review]
I agree with the feedback from Joe and Annie.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 3 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:12]: Pre-existing. Will change in separate PR.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 4 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:23]: Pre-existing. Will change in separate PR. (I've just split the single fragment into two).

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 5 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment]: Pre-existing. Will change in separate PR. (I've just split the single fragment into two).

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 6 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment]: Removed from this PR. Will be a follow-up.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 7 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment]: Removed from this PR. Will be a follow-up.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 8 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment]: Removed from this PR. Will be a follow-up.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 9 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment:19]: Pre-existing. Will be part of follow-up.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 10 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment]: Removed from this PR. Will be a follow-up.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 11 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment]: Removed from this PR. Will be a follow-up.

File: docs/src/main/sphinx/connector/jdbc-common-configurations.fragment (hunk 12 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/jdbc-common-configurations.fragment]: Removed from this PR.

**Comment by hashhar** [comment]
@jhlodin @electrum I've simplified this to just be a refactor that splits 1 file into two. I hope this is not controversial.

The new common JDBC configs (from BaseJdbcConfig etc.) can now start getting documented under `jdbc-common-configurations` fragment.

File: docs/src/main/sphinx/connector/phoenix.rst [hunk 13]
```diff
@@ -61,6 +61,8 @@ Property Name                                      Required   Description
                                                               connection properties.  These files must exist on the machines running Trino.
 ================================================== ========== ===================================================================================
 
+.. include:: jdbc-common-configurations.fragment
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/phoenix.rst:64]: This will be awkward in the context of this doc; The section above has a configuration properties table, introduced with "The following configuration properties are available. This fragment adds a second configuration properties table, introduced with "The following table describes general configuration properties for the connector."

Propose changing the fragment introduction to something like "The following table describes configuration properties that are available for this and other JDBC-based connectors" which at least distinguishes it a little bit from the above table.

Or, you can update the introduction on line 50 to say something like "The following Phoenix-specific configuration properties are available".

**Review by jhlodin — APPROVED** [review]
Mostly LGTM, but the Phoenix connector doc is confusing. Left a comment with suggestions.

File: docs/src/main/sphinx/connector/phoenix.rst (hunk 13 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/phoenix.rst:64]: +1

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 14]
```diff
@@ -115,6 +117,13 @@ Property Name               Default Value    Description
 Currently the connector only supports ``Log`` and ``MergeTree`` table engines
 in create table statement. ``ReplicatedMergeTree`` engine is not yet supported.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/clickhouse.rst:118]: ``CREATE TABLE``

File: docs/src/main/sphinx/connector/phoenix.rst (hunk 13 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/phoenix.rst:64]: The latter option makes more sense to me since the fragment applies to all JDBC connectors and `for this and other JDBC-based connectors` sounds weird when I'm looking at docs for a single JDBC connector.

Comment (hashhar) [reply docs/src/main/sphinx/connector/phoenix.rst:61]: Changed. PTAL. Also removed the double `Configuration` and `Configuration properties` redundancy in the Phoenix docs. This is how it looked earlier 
image

File: docs/src/main/sphinx/connector/clickhouse.rst (hunk 14 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/clickhouse.rst:118]: Pre-existing. Noted down, will fix in a later change. I'd like to keep this PR isolated to just mechanical splitting of things without changing unrelated things.

File: docs/src/main/sphinx/connector/phoenix.rst [hunk 15]
```diff
@@ -44,10 +44,7 @@ For HBase 2.x and Phoenix 5.x (5.1.0 or later) use:
 
     connector.name=phoenix5
 
-Configuration properties
-------------------------
-
-The following configuration properties are available:
+The following Phoenix-specific configuration properties are available:
```
Comment (rosewms) [inline docs/src/main/sphinx/connector/phoenix.rst:48]: I'd rephrase to remove "are available" it sounds awkward. I'd recommend something like:

>Review the available Phoenix-specific configuration properties:

**Review by mosabua — APPROVED** [review]
Works for me to split it up.

**Comment by hashhar** [comment]
Squashed and merged.

**Review by rosewms — APPROVED** [review]
LGTM
